### PR TITLE
fix: mini toc issues EXLM-293

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID:

Test URLs: https://jira.corp.adobe.com/browse/EXLM-293

- Before: https://main--exlm--adobe-experience-league.hlx.page/docs/authoring-guide-exl/using/markdown/cheatsheet
- After: https://bugfix-exlm-293--exlm--adobe-experience-league.hlx.page/docs/authoring-guide-exl/using/markdown/cheatsheet

### CODE REVIEW

File: blocks/mini-toc/mini-toc.js [hunk 1]
```diff
@@ -63,6 +66,22 @@ export default async function decorate(block) {
               highlight(false);
             }
           });
+          const anchor = anchors[0].parentElement;
```
Comment (thomsebastin) [inline blocks/mini-toc/mini-toc.js:69]: Maybe put a check here to ensure `anchor[0]` exists.

File: blocks/mini-toc/mini-toc.js [hunk 2]
```diff
@@ -63,6 +66,22 @@ export default async function decorate(block) {
...
+          const scrollableDiv = ctx.querySelector('.scrollable-div');
+          if (scrollableDiv) {
+            // dynamically make sure no item is partially visible
+            const anchorClientHeight = anchor.offsetHeight;
+            const anchorStylyes = getComputedStyle(anchor);
```
Comment (thomsebastin) [inline blocks/mini-toc/mini-toc.js:74]: there is a typo here, should be `anchorStyles`?

**Review by thomsebastin — APPROVED** [review]
Looks good. Left two comments to address.

**Comment by thomsebastin** [comment]
@nitin-rachabathuni Just a suggestion, it is always recommended to have same page in `before` and `after` section of PR description so that it's easy to verify the changes.

**Review by thomsebastin — CHANGES_REQUESTED** [review]
The TOC is not appearing on the page. Looks like the converter is sending only one section now. We need to fix this before we can merge.

File: blocks/mini-toc/mini-toc.js (hunk 1 above)
Comment (nitin-rachabathuni) [reply blocks/mini-toc/mini-toc.js:69]: We have ` if (anchors.length > 0) { ` check in line 31.

**Review by thomsebastin — APPROVED** [review]

**Pull request merged**